packages/x-data-grid/src/components/containers/GridRootStyles.ts

```diff
@@ -273,6 +280,15 @@ export const GridRootStyles = styled('div', {
     [`& .${c.columnHeader}:focus, & .${c.cell}:focus`]: {
       outline: `solid ${t.palette.primary.main} 1px`,
     },
+    [`&.${c['root--noToolbar']} [aria-rowindex="1"] [aria-colindex="1"]`]: {
+      borderTopLeftRadius: 'var(--unstable_DataGrid-radius)',
```

Comment:
```suggestion
      borderTopLeftRadius: 'calc(var(--unstable_DataGrid-radius) - 1px)',
```

There is a minor visual issue with using the same border radius for the element inside, in the screenshot below, you can see a gap between the curves:

<img width="106" alt="Screenshot 2024-07-30 at 09 14 51" src="https://github.com/user-attachments/assets/3e38beef-2ae6-450e-adb3-ae10a1191da4">

_I think_ we just want to minus a pixel based on this https://cloudfour.com/thinks/the-math-behind-nesting-rounded-corners/